packages/network-of-terms-reconciliation/src/query.ts

```diff
@@ -36,19 +38,41 @@ export async function reconciliationQuery(
       );
       const terms = termsResult instanceof Terms ? termsResult.terms : [];
       results[queryId] = {
-        result: terms.map(term => ({
-          id: term.id.value.toString(),
-          name: term.prefLabels.map(label => label.value).join(' • '), // Join similarly to network-of-terms-demo.
-          score: 1, // Hard-coded score for now because scoring will have to be done on our side because no scores are returned from the sources.
-          description: term.altLabels.map(label => label.value).join(' • '),
-        })),
+        result: terms
+          .map(term => ({
+            id: term.id.value.toString(),
+            name: term.prefLabels.map(label => label.value).join(' • '), // Join similarly to network-of-terms-demo.
+            score: score(queryString, term),
+            description: term.altLabels.map(label => label.value).join(' • '),
+          }))
+          .sort((a, b) => b.score - a.score),
       };
       return results;
     },
     Promise.resolve({})
   );
 }
 
+/**
+ * Calculate score for term based on case-insensitive Levenshtein distance between the query string and the term’s
+ * prefLabels.
+ */
+const score = (queryString: string, term: Term): number => {
+  if (term.prefLabels.length === 0) {
+    return 0;
+  }
+
+  return (
+    1 /
+    term.prefLabels.reduce(
```

Comment:
This only looks at prefLabels. Should we also look at other properties?

Reply:
Interesting thought! In my sample dataset I have persons named 'Jan Havicksz. Steen', 'Maxime François Antoine Lalanne' and 'Herman van Dodeweerd'. If I reconcile these with e.g. RKDartists our Recon API returns terms such as:

https://data.rkd.nl/artists/74809
prefLabel: Steen, Jan
altLabel: Steen, Jan Havicksz.

https://data.rkd.nl/artists/47489
prefLabel: Lalanne, Maxime
altLabel: Lalanne, Maxime François Antoine

https://data.rkd.nl/artists/2458
prefLabel: Armando
altLabel: Dodeweerd, Herman Dirk van

In other words: perfect matches, because of the altLabels. If we only calculate the scores based on prefLabels, the perfect matches may get a lower ranking. So it makes sense to also look at the altLabels. Perhaps we could calculate a score for both the prefLabel and the altLabel and use the highest score in the result?

Reply:
If that's a sound approach, we could also take the hiddenLabel into consideration.

Reply:
If that's a sound approach, we could also take the hiddenLabel into consideration.

Reply:
I have some hesitations about the hiddenLabel because that is not shown in the reconciliation results or preview, so it would become rather opaque to users what the score is based on.

Reply:
Yes, good point! On the other hand: that seems to be the exact application of `hiddenLabel`. From the SKOS primer [1]:

> A hidden lexical label, represented by means of the [skos:hiddenLabel](http://www.w3.org/TR/skos-reference#hiddenLabel) property, is a lexical label for a resource, where a KOS designer would like that character string to be accessible to applications performing text-based indexing and search operations, but would not like that label to be visible otherwise.

I could ask the Recon API testers what they think of it (i.e. should we look at hiddenLabel or not)?

[1] https://www.w3.org/TR/skos-primer/

Reply:
Please do! Just to be clear: the current (released) functionality does _not_ look at hiddenLabel for scoring. 

But because it does consider hiddenLabel to determine _whether to include a term in the search results_ in the first place, we can expect a problem with larger result sets, when terms that match well on hiddenLabel (but do not on their prefLabel/altLabel) get scored 0 and drop out of the results together. This problem will become even clearer when clients set a relatively low limit (e.g. 3).

Reply:
Thanks, very clear!